lib/rules/safe-context-keyword.js

```diff
@@ -50,16 +50,15 @@ module.exports.prototype = {
 
         // var that = this
         file.iterateNodesByType('VariableDeclaration', function(node) {
-            var firstToken = node.firstToken;
-
-            // Miss destructing assignment (#1699)
-            if (file.getNextToken(firstToken).value === '{') {
-                return;
-            }
 
             for (var i = 0; i < node.declarations.length; i++) {
                 var decl = node.declarations[i];
 
+                // Miss destructing assignment (#1699, #2119)
+                if (file.getNextToken(file.getPrevToken(decl)).value === '{') {
```

Comment:
Can't port that to the `2.x` branch though, so it might have to wait until 3.0 release :/